### PR TITLE
django debug toolbar for better profiling and debugging

### PR DESCRIPTION
**Pull request by Madoshakalaka**
`django-debug-toolbar` is a PYPI package that provides great UI for profiling and debugging.

It shows as a hovering green badge in development only.

green-icon

(powerful sql profiling, I will use this to debug performance issues)
powerful sql profiling

The toolbar is capable of many things:

- You can drag it around
- You can expand and collapse it
- It shows varaibles in settings.py, shows context variables used in the rendering of each template
- shows SQL qeuries and their time taken
- shows logging messages
- does profiling if you want
- highly configurable through settings.py

### CODE REVIEW

File: CreeDictionary/CreeDictionary/settings.py [hunk 1]
```diff
@@ -64,6 +65,17 @@
...
+if DEBUG:
+    MIDDLEWARE.insert(0, "debug_toolbar.middleware.DebugToolbarMiddleware")
+
+    # works with django-debug-toolbar app
+    DEBUG_TOOLBAR_CONFIG = {
+        # Toolbar options
+        "SHOW_COLLAPSED": True, # collapse the toolbar by default
+    }
+
+INTERNAL_IPS = ["127.0.0.1"]
```
Comment (Madoshakalaka) [inline CreeDictionary/CreeDictionary/settings.py]: django debug toolbar settings

File: CreeDictionary/DatabaseManager/test_db_builder.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 from tqdm import tqdm
 
-from profiling.timing import timed
+from utils.profiling import timed
```
Comment (Madoshakalaka) [inline CreeDictionary/DatabaseManager/test_db_builder.py]: old `profiling` directory is deleted except for one utility - the `timed` decorator.

File: CreeDictionary/tests/utils_tests/test_profiling.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import time
+
+from utils.profiling import timed
+
+
+def test_timed_decorator(capsys):
```
Comment (Madoshakalaka) [inline CreeDictionary/tests/utils_tests/test_profiling.py]: new test for coverage

File: docs/developers-guide.md [hunk 4]
```diff
@@ -151,7 +151,7 @@ It recognizes the following:
 
  - The Django settings module in `setup.cfg` (for `pytest-django` to work)
  - `--doctest-modules` `--mypy` in `Pipfile [script]` (to enable doctest and Mypy tests)
- - `DEBUG=False` in `.env`
+ - `DEBUG=False` `USE_TEST_DB=True` in `.env`
```
Comment (Madoshakalaka) [inline docs/developers-guide.md]: outdated documentation

File: CreeDictionary/CreeDictionary/settings.py [hunk 5]
```diff
@@ -64,6 +64,26 @@
...
+            0, "debug_toolbar.middleware.DebugToolbarMiddleware"
+        )  # middleware order is important
+
+        # works with django-debug-toolbar app
+        DEBUG_TOOLBAR_CONFIG = {
+            # Toolbar options
+            "SHOW_COLLAPSED": True,  # collapse the toolbar by default
+        }
+
+        INTERNAL_IPS = ["127.0.0.1"]
```
Comment (eddieantonio) [inline CreeDictionary/CreeDictionary/settings.py]: 👍

File: CreeDictionary/tests/utils_tests/test_profiling.py (hunk 3 above)
Comment (eddieantonio) [reply CreeDictionary/tests/utils_tests/test_profiling.py]: quick napffss

File: docs/developers-guide.md [hunk 6]
```diff
@@ -167,17 +167,10 @@ Or, for interactive use:
 Profiling Code
 --------------
 
-    pipenv run profile <script_name>
+We use PYPI utility [django-toolbar-toolbar](https://django-debug-toolbar.readthedocs.io/en/latest/).
```
Comment (eddieantonio) [inline docs/developers-guide.md]:
```suggestion
We use [django-toolbar-toolbar](https://django-debug-toolbar.readthedocs.io/en/latest/). 
```

File: docs/developers-guide.md [hunk 7]
```diff
@@ -167,17 +167,10 @@ Or, for interactive use:
 Profiling Code
 --------------
 
-    pipenv run profile <script_name>
+We use PYPI utility [django-toolbar-toolbar](https://django-debug-toolbar.readthedocs.io/en/latest/). 
+It provides great UI for stack tracing and sql query profiling.
```
Comment (eddieantonio) [inline docs/developers-guide.md]:
```suggestion
It provides great UI for stack tracing and SQL query profiling.
```

File: setup.cfg [hunk 8]
```diff
@@ -8,3 +8,9 @@ ignore_missing_imports = True
 
 [mypy-API.migrations.*]
 ignore_errors = True
+
+[coverage:run]
+omit =
+  CreeDictionary/CreeDictionary/settings.py
+  CreeDictionary/manage.py
+  CreeDictionary/setup.py
```
Comment (eddieantonio) [inline setup.cfg]: Haha, fair! 👍

**Comment by Madoshakalaka** [comment]
> Feel free to commit or ignore my documentation nitpicks.

I'm Chinese.

**Pull request merged**